src/test/java/io/managed/services/test/cluster/DataPlaneClusterTest.java

```diff
@@ -0,0 +1,359 @@
+ * <ul>
+ *     <li> testReservedDeploymentExistence: reserved deployment exists (developer and standard instances)
+ *     <li> testReservedDeploymentDoesNotPreventKafkaCreation: reserved deployment is used to create actual kafka instance (developer and standard instances)
+ *     <li> testStandardKafkaNodeAutoscaling: machineset resource is scaled when new instances are created (standard instance as developer are already at min max nodes value)
+ *     <li> testCapacityReportingUpdated: managedkafkaagent reflect creation of new kafka instance in its remaining capacity (developer and standard instance)
+ * </ul>
+ * <p>
+ * <b>Requires:</b>
+ * <ul>
+ *     <li> PRIMARY_USERNAME
+ *     <li> PRIMARY_PASSWORD
+ * </ul>
+ */
+@Log4j2
+public class DataPlaneClusterTest extends TestBase {
+
+    static final String KAFKA_INSTANCE_NAME = "cl-e2e-" + Environment.LAUNCH_KEY;
+    private OpenShiftClient oc;
+
+    private static int maxStandardInstancesInCluster;
+    private static int maxDeveloperInstancesInCluster;
+    private static final String KAFKAS_MGMT_21_CODE = "KAFKAS-MGMT-21";
+    private static final String PLAN_STANDARD = "standard.x1";
+    private static final String INSTANCE_NAME_PREFIX = "cl-e2e-placeholder-";
+    private KafkaMgmtApi kafkaMgmtApi;
+
+    @BeforeClass
+    @SneakyThrows
+    public void bootstrap() {
+
+        log.info("build config");
+        Config config = new ConfigBuilder()
+            .withMasterUrl("https://api.mk-stage-0622.bd59.p1.openshiftapps.com:6443")
+            .withOauthToken(Environment.PROMETHEUS_WEB_CLIENT_ACCESS_TOKEN)
+            .withTrustCerts(true)
+            .build();
+
+        log.info("init openshift client");
+        oc = new KubernetesClientBuilder().withConfig(config).build().adapt(OpenShiftClient.class);
+
+        log.info("init cluster capacity info");
+        maxStandardInstancesInCluster = FleetshardUtils.getClusterCapacityFromMKAgent(oc, ManagedKafkaType.standard);
+        log.debug("Max standard instances according to MKAgent capacity '{}'", maxStandardInstancesInCluster);
+        maxDeveloperInstancesInCluster = FleetshardUtils.getClusterCapacityFromMKAgent(oc, ManagedKafkaType.standard);
+        log.debug("Max developer instances according to MKAgent capacity '{}'", maxDeveloperInstancesInCluster);
+
+        log.info("init kafka management");
+        var apps = ApplicationServicesApi.applicationServicesApi(
+            Environment.PRIMARY_USERNAME,
+            Environment.PRIMARY_PASSWORD);
+        kafkaMgmtApi = apps.kafkaMgmt();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void teardown() {
+
+        // delete kafka instance
+        try {
+            KafkaMgmtApiUtils.cleanKafkaInstance(kafkaMgmtApi, KAFKA_INSTANCE_NAME);
+        } catch (Throwable t) {
+            log.error("clean main kafka instance error: ", t);
+        }
+    }
+
+    @DataProvider(name = "managedKafkaTypes")
+    public Object[][] managedKafkaTypeDataProvider() {
```

Comment:
```suggestion
    public ManagedKafkaType[][] managedKafkaTypeDataProvider() {
```

can we use the specific data type?

Reply:
yeah it just ended 50%/50% with conversion, but this would make it more sense (assuming type will remain as such, implicit conversion on some places would favour your suggestion)